testcontainers/src/clients/cli.rs

```diff
@@ -7,30 +7,39 @@ use std::{
     collections::HashMap,
     ffi::{OsStr, OsString},
     process::{Child, Command, Stdio},
-    sync::{Arc, RwLock},
+    sync::{OnceLock, RwLock},
     thread::sleep,
     time::{Duration, Instant},
 };
 
 const ONE_SECOND: Duration = Duration::from_secs(1);
 const ZERO: Duration = Duration::from_secs(0);
 
-/// Implementation of the Docker client API using the docker cli.
-///
-/// This (fairly naive) implementation of the Docker client API simply creates `Command`s to the `docker` CLI. It thereby assumes that the `docker` CLI is installed and that it is in the PATH of the current execution environment.
-#[derive(Debug)]
-pub struct Cli {
-    inner: Arc<Client>,
+static CLI_DOCKER: OnceLock<Client> = OnceLock::new();
+pub(crate) fn docker_client() -> &'static Client {
+    CLI_DOCKER.get_or_init(|| Client::new::<env::Os>())
 }
 
-impl Cli {
-    pub fn run<I: Image>(&self, image: impl Into<RunnableImage<I>>) -> Container<'_, I> {
-        let image = image.into();
+pub trait RunViaCli<I: Image> {
+    fn start(self) -> Container<I>;
+}
 
-        if let Some(network) = image.network() {
-            if self.inner.create_network_if_not_exists(network) {
-                let mut guard = self
-                    .inner
+impl<I> RunViaCli<I> for I
+where
+    I: Image,
+    I::Args: Default,
+{
+    fn start(self) -> Container<I> {
```

Comment:
This is an helper implementation that allow you to start container directly from Image without needing to wrap image with RunnableImage::from